javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -807,23 +809,33 @@ private static void addSetterMethod(IType propertyWireType, IType propertyClient
         methodBlock.methodReturn("this");
     }
 
-    private static void generateHeaderCollectionDeserialization(ClientModelProperty property, JavaBlock block) {
-        int headerPrefixLength = property.getHeaderCollectionPrefix().length();
-        // HeaderCollections are always Maps that use String as the key.
-        MapType wireType = (MapType) property.getWireType();
-        String collectionName = property.getName() + "HeaderCollection";
+    private static void generateHeaderCollectionDeserialization(List<ClientModelProperty> properties, JavaBlock block) {
+        for (ClientModelProperty property : properties) {
+            // HeaderCollections are always Maps that use String as the key.
+            MapType wireType = (MapType) property.getWireType();
+
+            // Prefix the map with the property name for the cases where multiple header collections exist.
+            block.line("%s %sHeaderCollection = new HashMap<>();", wireType, property.getName());
+        }
 
-        // Prefix the map with the property name for the cases where multiple header collections exist.
-        // TODO: If there are multiple header collections they can be combined into a single loop over the headers.
-        // Not a high priority as loop should be relatively cheap, and only constant time.
-        block.line("%s %s = new HashMap<>();", wireType, collectionName);
         block.line();
+
         block.block("for (HttpHeader header : rawHeaders)", body -> {
-            body.ifBlock(String.format("!header.getName().startsWith(\"%s\")", property.getHeaderCollectionPrefix()),
-                ifBlock -> ifBlock.line("continue;"));
-            body.line(collectionName + ".put(header.getName().substring(%d), header.getValue());", headerPrefixLength);
+            for (ClientModelProperty property : properties) {
+                body.ifBlock(String.format("header.getName().startsWith(\"%s\")", property.getHeaderCollectionPrefix()),
+                    ifBlock -> {
+                        ifBlock.line("%sHeaderCollection.put(header.getName().substring(%d), header.getValue());",
+                            property.getName(), property.getHeaderCollectionPrefix().length());
+                        ifBlock.line("break;");
+                    });
+            }
```

Comment:
Do we have an example/testcase of what was previously generated vs what is generated now? The generated code here is slightly different and I just wanted to check if the two are equivalent.

Reply:
There isn't a case in the generation Swagger but Storage Blobs has a case where there are multiple header collection in the same response.

Reply:
It may be good to add the before/after code snippets in the PR description. I can create an issue in testserver to add a test case for this.